linux/LibSensors.c

```diff
@@ -261,6 +261,43 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
       goto out;
    }
 
+   /* Read sysfs to get HT/SMT cores and correctly distribute the temperature values */
+   if (coreTempCount < activeCPUs) {
+      double oldData[existingCPUs + 1];
+      memcpy(oldData, data, sizeof(oldData));
+
+      int dataCell = 1;
+      for (unsigned int i = 0; i < existingCPUs; i++) {
+         char path[100];
+         sprintf(path, "/sys/devices/system/cpu/cpu%d/topology/thread_siblings_list", i);
+         FILE* file = fopen(path, "r");
+         if (file == NULL) {
+            data[i+1] = NAN;
+            dataCell++;
+            continue;
+         }
+
+         char contents[10];
+         fgets(contents, 10, file);
+         fclose(file);
+
+         int start, end;
+         int rangeNums = sscanf(contents, "%d-%d", &start, &end);
```

Comment:
This buffer is too short for the potential values that could be returned by the kernel. Should be at least 24 bytes (2x 10 bytes for `int32_t` plus 2 bytes for `-` and `\n`, plus one final `\0` terminator.